app/main.py

```diff
@@ -2,0 +1,28 @@
+class Animal:
+    alive = []
+
+    def __init__(self, name: str, health: int = 100,) -> None:
+        self.name = name
+        self.health = health
+        self.hidden = False
+        Animal.alive.append(self)
+
+    def __repr__(self) -> None:
+        return (f"{{Name: {self.name}, "
+                f"Health: {self.health}, Hidden: {self.hidden}}}")
+
+
+class Herbivore(Animal):
+    def hide(self) -> bool:
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+
+    def bite(self, animal: None) -> None:
```

Comment:
The `bite` method's parameter `animal` is typed as `None`, which is incorrect. It should be typed as `Animal` or `Herbivore` to reflect the expected input.